File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/java/python/codeforces_189_A.py

import sys

def main():
    import sys
    import sys
    from sys import stdin
    
    input = stdin.read().split()
    n = int(input[0])
    maxn = 4010
    a = int(input[1])
    b = int(input[2])
    c = int(input[3])
    arr = [a, b, c]
    dp = [float('-inf')] * maxn
    dp[0] = 0
    for i in range(3):
        for j in range(arr[i], len(dp)):
            dp[j] = max(dp[j], dp[j - arr[i]] + 1)
    print(dp[n])

if __name__ == "__main__":
    main()

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/codeforces_189_A.java

import java.util.Arrays;
import java.util.Scanner;

public class codeforces_189_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int[] arr = new int[input.length];
        for (int i = 0; i < input.length; i++) {
            arr[i] = Integer.parseInt(input[i]);
        }
        int length = arr[0];
        int[] subArray = Arrays.copyOfRange(arr, 1, 4);
        Arrays.sort(subArray);
        if (length % subArray[0] == 0) {
            System.out.println(length / subArray[0]);
            return;
        }
        int ans = 0;
        if (subArray[0] == subArray[1]) {
            int k = 0;
            for (int i = 0; i <= length / subArray[2]; i++) {
                int check = length - i * subArray[2];
                if (check >= 0 && check % subArray[0] == 0) {
                    k = check / subArray[0];
                    ans = Math.max(ans, k + i);
                }
            }
        } else {
            int k = 0;
            for (int i = 0; i <= length / subArray[2]; i++) {
                for (int j = 0; j <= length / subArray[1]; j++) {
                    int check = length - i * subArray[2] - j * subArray[1];
                    if (check >= 0 && check % subArray[0] == 0) {
                        k = check / subArray[0];
                        ans = Math.max(ans, k + i + j);
                    }
                }
            }
        }
        System.out.println(ans);
    }
}

